### DISCUSSION TITLE
Missing Focusable property

### DISCUSSION THREAD

**Discussion by fabianoriccardi** [microsoft/microsoft-ui-xaml#9814] [Q&A]
I have noted that WPF offers the property UIElement.Focusable which helps to discriminate (and set) quickly the controls that can receive the focus, but in WinUI there is not such feature. This can especially useful when building custom components.

Will WASDK implement Focusable property as WPF? Currently, does an alternative exist to this property? 

I have read the following pages but I didn't find the cohemprensive answer:

- WASDK: Focus() is part of UIElement https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.uielement.focus?view=windows-app-sdk-1.5 
- UWP: Focus() is part of control https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.focus?view=winrt-26100 
- WPF (Windows Desktop 8): Focus() is part of UIElement and it has the property Focusable https://learn.microsoft.com/en-us/dotnet/api/system.windows.uielement.focus?view=windowsdesktop-8.0

**Comment by whiskhub** [comment]
It's not as nice, but maybe `FocusManager.FindNextElement()` can help? 
https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.focusmanager.findnextelement?view=windows-app-sdk-1.5

For setting it, maybe `IsTabStop` and `IsEnabled` can help. But I'm afraid they are not 1:1 mappings to the WPF functionality.

**Reply by fabianoriccardi** [reply]
Thanks, `IsTabStop` can help, at least for the keyboard interaction. And, as expected, the TextBlock overrides the default value (true) to false.

However, there should be a method to block a control to get the focus, since the TextBlock doesn't get the focus when I click on it.
In WPF, `Focusable` property was used to enable/disable the focus for both keyboard and mouse interaction. [Here](https://learn.microsoft.com/en-us/answers/questions/686424/why-wont-the-textblock-accept-the-focus) an example.

EDIT: read my next comment

**Reply by fabianoriccardi** [reply]
I have tested a bit more the `IsTabStop`, and it also blocks the GettingFocus and the other events even on mouse interaction. So, basically the `IsTabStop` seems the new `Focusable` property.

If this is true and I'm not missing some other details, the documentation of `IsTabStop` should specify that it stops the UIElement from getting the Focus for both mouse and interaction (it may help the ones who are migrating from WPF).
